dwave/system/testing.py

```diff
@@ -117,69 +224,239 @@ def __init__(self, broken_nodes=None, broken_edges=None,
+                                    'reverse_annealing_without_reinit_delay_time_delta': -1.5,
+                                    'default_readout_thermalization': 0.0,
+                                    'decorrelation_max_nominal_anneal_time': 2000.0,
+                                    'decorrelation_time_range': [500.0, 10000.0]},
+            'programming_thermalization_range': [0.0, 10000.0],
+            'readout_thermalization_range': [0.0, 10000.0],
+            'tags': [],
+            'category': 'qpu',
+            'quota_conversion_rate': 1})
+        
+        if properties is not None:
+            # provided properties overwrite defaults, note that
+            # particular care should be taken with respect to
+            # topology-dependent arguments:
+            self.properties.update(properties)
+        
+    @classmethod
+    def from_qpu_sampler(cls, sampler):
+        return cls(properties=sampler.properties,nodelist=sampler.nodelist,edgelist=sampler.edgelist)
+
     @dimod.bqm_structured
-    def sample(self, bqm, num_reads=1, flux_biases=[], **kwargs):
-        # we are altering the bqm if flux_biases given
+    def sample(self, bqm, **kwargs):
+        
+        # Check kwargs compatibility with parameters and substitute sampler:
+        mocked_parameters={'answer_mode',
+                           'max_answers',
+                           'num_reads',
+                           'label'}
+        if SubstituteSampler is not dimod.SimulatedAnnealingSampler:
+            mocked_parameters.add('initial_state')
+            # steepest greedy descent is the best substitute to
+            # standardize upon. A few additional considerations as
+            # dwave-greedy is modified and processor scale grows:
+            # (1) For large sample sets, it would be more efficient to
+            # handle ``answer_mode`` and ``max_answers`` within the
+            # lower-level optimized code - just as they are handled server
+            # side in QPU calls.
+            # (2) We could also consider using 'large_sparse_opt' to
+            # exploit fixed connectivity at large scale for current QPU
+            # designs, but unlikely to be a significant inefficiency.
+            
+            # The fallback sampler (SA) is an inferior choice, but
+            # greedy may not always be installed, unlike dimod.
+        for kw in kwargs:
+            if kw in self.parameters:
+                if self.parameter_warnings and kw not in mocked_parameters:
+                    warnings.warn(f'{kw!r} parameter is valid for DWaveSampler(), '
+                                  'but not mocked in MockDWaveSampler().')
+            else:
+                raise ValueError(f'kwarg {kw!r} invalid for MockDWaveSampler()')
 
-        info = dict(problem_id=str(uuid4()))
+        # Timing values are for demonstration only. These could be made
+        # adaptive to sampler parameters and mocked topology in principle.
```

Comment:
Yes, we should do it in a follow-up PR, using `estimate_qpu_access_time`, once https://github.com/dwavesystems/dwave-cloud-client/pull/530 is merged.